From 3992ed5725a8d89196338eec40826c9245173195 Mon Sep 17 00:00:00 2001
From: Abhishek <apatgaonkar@habana.ai>
Date: Fri, 15 Mar 2024 14:00:36 +0200
Subject: [PATCH] [FSW-402] fs1: security_init: add jtag handle stub

This patch adds stub for security init and jtag
enable/disable.

Change-Id: I0b6463c238e1cf9d8b7b7100cc982f614dbcf869
---

diff --git a/app/boot_rom.c b/app/boot_rom.c
index 1aa63c6..a4f4eaa 100644
--- a/app/boot_rom.c
+++ b/app/boot_rom.c
@@ -5,6 +5,7 @@
  */
 
 #include <boot_rom.h>
+#include <security.h>
 
 struct boot_rom_global_s __attribute__((section (".rom_persistent_section")))boot_rom_global = {
 	.config_fuse = 0,
@@ -23,5 +24,7 @@
 {
 	boot_rom_init();
 
+	boot_rom_security_init();
+
 	return 0;
 }
diff --git a/common/security.c b/common/security.c
new file mode 100644
index 0000000..47ff6c3
--- /dev/null
+++ b/common/security.c
@@ -0,0 +1,18 @@
+// SPDX-License-Identifier:     GPL-2.0+
+/*
+ * Copyright (C) 2024 HabanaLabs Ltd.
+ * All Rights Reserved.
+ *
+ */
+
+static void security_handle_jtag_if(void)
+{
+	/* TODO: add jtag enable/disable logic */
+}
+
+void boot_rom_security_init(void)
+{
+	/* Based on efuse enable/disable jtag interface and clear FSM timer */
+	security_handle_jtag_if();
+}
+
diff --git a/common/security.h b/common/security.h
new file mode 100644
index 0000000..cfdf06a
--- /dev/null
+++ b/common/security.h
@@ -0,0 +1,11 @@
+/* SPDX-License-Identifier:     GPL-2.0+
+ *
+ * Copyright (C) 2024 HabanaLabs Ltd.
+ */
+
+#ifndef _SECURITY_H_
+#define _SECURITY_H_
+
+void boot_rom_security_init(void);
+
+#endif /* _SECURITY_H_ */
